web/crux/src/app/pipeline/pipeline.service.ts

```diff
@@ -0,0 +1,304 @@
+import { Injectable, Logger } from '@nestjs/common'
+import { JwtService } from '@nestjs/jwt'
+import { Identity } from '@ory/kratos-client'
+import { Subject } from 'rxjs'
+import {
+  AZURE_DEV_OPS_STATE_CHANGED_EVENT_TYPE_VALUES,
+  AzureDevOpsCredentials,
+  AzureDevOpsHook,
+  AzureHook,
+  AzureRepository,
+  AzureTrigger,
+  PipelineRunStatusEvent,
+} from 'src/domain/pipeline'
+import { PipelineTokenPayload } from 'src/domain/pipeline-token'
+import { generateNonce } from 'src/domain/utils'
+import { CruxBadRequestException } from 'src/exception/crux-exception'
+import AzureDevOpsService from 'src/services/azure-devops.service'
+import EncryptionService from 'src/services/encryption.service'
+import PrismaService from 'src/services/prisma.service'
+import TeamRepository from '../team/team.repository'
+import {
+  AzurePipelineStateDto,
+  CreatePipelineDto,
+  PipelineDetailsDto,
+  PipelineDto,
+  PipelineRunDto,
+  TriggerPipelineDto,
+  UpdatePipelineDto,
+} from './pipeline.dto'
+import PipelineMapper from './pipeline.mapper'
+
+@Injectable()
+export default class PipelineService {
+  private readonly logger = new Logger(PipelineService.name)
+
+  readonly runStatusEvent = new Subject<PipelineRunStatusEvent>()
+
+  constructor(
+    private readonly teamRepository: TeamRepository,
+    private readonly prisma: PrismaService,
+    private readonly mapper: PipelineMapper,
+    private readonly azureService: AzureDevOpsService,
+    private readonly jwtService: JwtService,
+    private readonly encryptionService: EncryptionService,
+  ) {}
+
+  async getPipelines(teamSlug: string): Promise<PipelineDto[]> {
+    const pipelines = await this.prisma.pipeline.findMany({
+      where: {
+        team: {
+          slug: teamSlug,
+        },
+      },
+      include: {
+        runs: {
+          orderBy: {
+            createdAt: 'desc',
+          },
+          take: 1,
```

Comment:
If you use `take: 1`, it will return only the first one, right? However, with respect to the function, this is not what you want.

Reply:
This is being used for the pipeline list, and we only need the last run, so we can display its status.